### PR TITLE
03/14 dbt-core 1.4.5 & Other Dependency Updates

### PR DESCRIPTION
**Pull request by jlarue26**
### Summary

Pull in all updated dependencies identified by Dependabot, including dbt-core 1.4.5.

### Description

Merged all of the PRs created by Dependabot into one branch so everything can be tested at once.

### Test Results

- [x] `./.github/scripts/smoke_test.sh` - all pass
- [x] `./tests/*.*` - all pass

### Changelog

-   [x] Added a summary of what this PR accomplishes to CHANGELOG.md

### Related Issue

None.

### CODE REVIEW

**Comment by jlarue26** [comment]
when running `pip install -r ./dbt-dremio/dev_requirements.txt`, I received the following:

> ERROR: Cannot install agate==1.7.1 and dbt-core==1.4.5 because these package versions have conflicting dependencies.
> 
> The conflict is caused by:
>     The user requested agate==1.7.1
>     dbt-core 1.4.5 depends on agate<1.7.1 and >=1.6
> 
> To fix this you could try to:
> 1. loosen the range of package versions you've specified
> 2. remove package versions to allow pip attempt to solve the dependency conflict

I went with option 2 and removed the explicit version of `agate` from `dev_requirements.txt`. I also removed the explicit version of but left `dbt-core==1.4.5`.          

I uninstalled `agate`, `dbt-core`, and `dbt-tests-adapter`. Re-running `pip install -r ./dbt-dremio/dev_requirements.txt` correctly installed `agate` v1.7.0 and `dbt-tests-adapter` v1.4.5.

**Review by ArgusLi — DISMISSED** [review]
Looks good to me.

**Comment by jlarue26** [comment]
[dbt-dremio_test_results_pr_156_dependency_updates_20230320_165305.txt](https://github.com/dremio/dbt-dremio/files/11024043/dbt-dremio_test_results_pr_156_dependency_updates_20230320_165305.txt)

**Comment by ravjotbrar** [comment]
I believe you still need to change the dbt versions in the third party licenses file.

**Comment by jlarue26** [comment]
@ravjotbrar good catch. Updated.
I used a virtual environment to ensure dve_requirements and third party licenses are correct.

**Review by ravjotbrar — APPROVED** [review]
Looks good!

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# dbt-dremio 1.4.1 - release TBD
+# dbt-dremio 1.4.5 - release TBD
```
Comment (jlarue26) [inline CHANGELOG.md:1]:
```suggestion
# dbt-dremio 1.4.5 - release March 23, 2023
```

**Pull request merged**